src/bst/core.clj

```diff
@@ -31,99 +31,50 @@
-      (< (factor tree) -1)))
-
-(defn is-left-right-case?
-  "Returns true if right sub tree of left child is imbalanced else false"
-  [tree]
-  (if (empty? tree) false
-      (and (is-left-case? tree) (< (factor (:left tree)) 0))))
-
-(defn is-left-left-case?
-  "Returns true if left sub tree of left child is imbalanced else false"
-  [tree]
-  (if (empty? tree) false
-      (and (is-left-case? tree) (> (factor (:left tree)) 0))))
-
-(defn is-right-right-case?
-  "Returns true if right sub tree of right child is imbalanced else false"
-  [tree]
-  (if (empty? tree) false
-      (and (is-right-case? tree) (< (factor (:right tree)) 0))))
-
-(defn is-right-left-case?
-  "Returns true if right sub tree of left child is imbalanced else false"
-  [tree]
-  (if (empty? tree) false
-      (and (is-right-case? tree) (> (factor (:right tree)) 0))))
-
-(defn rotate-left
-  "Returns the left rotated tree "
-  [{:keys [root left right] :as tree}]
-  (let [pivot (:root right)
-        left-pivot (:left right)
-        right-pivot (:right right)]
-    (if (empty? tree) {}
-        {:root pivot
-         :left {:root root
-                :left left
-                :right left-pivot}
-         :right right-pivot})))
-
-(defn rotate-right
-  "Returns the right rotated tree"
-  [{:keys [root left right] :as tree}]
-  (let [pivot (:root left)
-        left-pivot (:left left)
-        right-pivot (:right left)]
-    (if (empty? tree) {}
-        {:root pivot
-         :left left-pivot
-         :right {:root root
-                 :left right-pivot
-                 :right right}})))
-
-(defn balance
-  "Returns a balanced bst"
-  [{:keys [left right] :as tree}]
-  (cond
-    (is-right-left-case? tree) (rotate-left
-                                (assoc tree :right (rotate-right right)))
-
-    (is-left-right-case? tree) (rotate-right
-                                (assoc tree :left (rotate-left left)))
-
-    (is-right-right-case? tree) (rotate-left tree)
-
-    (is-left-left-case? tree) (rotate-right tree)
-
-    :else tree))
-
 (defn insert-node
   "Returns a bst after inserting a new node"
   [{:keys [root] :as tree} value]
   (cond
     (nil? root) {:root value :left nil :right nil}
-    (neg? (compare value root)) (balance
-                                 (update tree :left insert-node value))
-    (pos? (compare value root)) (balance
-                                 (update tree :right insert-node value))
+    (neg? (compare value root)) (update tree :left insert-node value)
+    (pos? (compare value root)) (update tree :right insert-node value)
     :else tree))
+(declare create)
+(defn inorder-traversal
+  [{:keys [root left right] :as tree}]
+  (cond (empty? tree) []
+        (some? tree)  (concat (inorder-traversal left)
+                              [root]
+                              (inorder-traversal right))))
+
+(defn traverse-and-balance
+  "Returns a balanced bst given a sorted list of node values"
+  ([nodes start  end]
```

Comment:
- Do you need the enclosing `()` if there is just one arity?
- There are 2 spaces between `start` and `end`.